secant/Features/BackupFlow/RecoveryPhraseValidation.swift

```diff
@@ -0,0 +1,232 @@
+    }
+
+    /// drives the state machine represented on this Enum by the action of applying a chip into a group of words containing an empty slot that has to be completed
+    func apply(chip: PhraseChip.Kind, into group: Int) -> Self {
+        guard case let PhraseChip.Kind.unassigned(word) = chip else { return self }
+
+        switch self.step {
+        case .initial:
+            guard let missingChipIndex = missingWordChips.firstIndex(of: chip) else { return self }
+
+            var newMissingWords = missingWordChips
+            newMissingWords[missingChipIndex] = .empty
+
+            return RecoveryPhraseValidationState(
+                phrase: phrase,
+                missingIndices: missingIndices,
+                missingWordChips: newMissingWords,
+                validationWords: [ValidationWord(groupIndex: group, word: word)]
+            )
+
+        case .incomplete:
+            guard let missingChipIndex = missingWordChips.firstIndex(of: chip) else { return self }
+
+            if validationWords.count < (RecoveryPhraseValidationState.phraseChunks - 1) {
+                var newMissingWords = missingWordChips
+                newMissingWords[missingChipIndex] = .empty
+
+                var newCompletionState = Array(validationWords)
+                newCompletionState.append(ValidationWord(groupIndex: group, word: word))
+
+                return RecoveryPhraseValidationState(
+                    phrase: phrase,
+                    missingIndices: missingIndices,
+                    missingWordChips: newMissingWords,
+                    validationWords: newCompletionState
+                )
+            } else {
+                var newCompletion = validationWords
+                newCompletion.append(ValidationWord(groupIndex: group, word: word))
+
+                return RecoveryPhraseValidationState(
+                    phrase: phrase,
+                    missingIndices: missingIndices,
+                    missingWordChips: Array(repeating: .empty, count: RecoveryPhraseValidationState.phraseChunks),
+                    validationWords: newCompletion
+                )
+            }
+        case .complete:
+            return self
+        }
+    }
+
+    static func randomIndices() -> [Int] {
+        return (0..<phraseChunks).map { _ in
+            Int.random(in: 0 ..< wordGroupSize)
+        }
+    }
+}
+
+extension RecoveryPhrase.Group {
+    /// Returns an array of words where the word at the missing index will be an empty string
+    func words(with missingIndex: Int) -> [String] {
+        assert(missingIndex >= 0)
+        assert(missingIndex < self.words.count)
+        var wordsApplyingMissing = self.words
+        wordsApplyingMissing[missingIndex] = ""
+        return wordsApplyingMissing
+    }
+}
+
+enum RecoveryPhraseValidationAction: Equatable {
+    case updateRoute(RecoveryPhraseValidationState.Route?)
+    case reset
+    case drag(wordChip: PhraseChip.Kind, intoGroup: Int)
```

Comment:
Suggestion:
Instead of naming things after UI implementation details, we keep the naming generic so that if the UI changes we don't have a store that is out of date with how the user interaction works.

I would consider naming this `move` so that it doesn't have anything to do with the term "drag and drop".